mne_bids/path.py

```diff
@@ -1237,14 +1233,15 @@ def _parse_ext(raw_fname):
     return fname, ext
 
 
-def _infer_datatype_from_path(fname):
+def _infer_datatype_from_path(fname: Path):
```

Comment:
this function is used by `get_bids_path_from_fname()`

IMHO it's doing too much magic too, but here I only ensured it keeps on working